forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth
+
+import okhttp3.OkHttpClient
+
+/**
+ * Manages SDK configuration information
+ */
+data class FROptions(val server: Server,
+                     val oauth: OAuth,
+                     val service: Service,
+                     val urlPath: UrlPath,
+                     val sslPinning: SSLPinning,
+                     val logger: Log) {
+
+    companion object {
+        @JvmStatic
+        fun equals(old: FROptions?, new: FROptions?): Boolean {
```

Comment:
This comparison function is different from the isConfigChanged from the ConfigHelper.kt

Why do we have 2 different comparisons? I think this is a protocol one, but in general they should be aligned in the things they compare. We might need to align iOS as well here so we treat the FROptions equality the same in both platforms

Reply:
Unless I am missing something, this "equals", could be used to check if we treat an FRoptions object as "changed" or not based on the rules we have defined. Does it have some extra usage, that requires to check all of the values?

Reply:
nope the purpose is different, there are two types of comparision we are doing , this checks for the local cached options and other is comparing the value from sharedpreference , they are built for different reasons. i am gonna change the method name of isConfigChanged to isConfigDifferentFromPersistedValue .